File: PeppaAnimations/app/src/main/java/rob/myappcompany/peppaanimations/MainActivity.java
package rob.myappcompany.peppaanimations;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    public void fade(View view){

        Log.i("Info", "ImageView Tapped");

        ImageView imageView = (ImageView) findViewById(R.id.imageView);

        imageView.animate().scaleX(0.5f).scaleY(0.5f).setDuration(1000);

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView imageView = (ImageView) findViewById(R.id.imageView);

        imageView.setX(-1000);

        imageView.animate().translationXBy(1000).rotation(3600).setDuration(2000);

    }
}

File: Snapchat/app/src/main/java/rob/myappcompany/snapchat/MainActivity.kt
package rob.myappcompany.snapchat

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.fragment.app.FragmentActivity
import com.google.firebase.auth.FirebaseAuth
import rob.myappcompany.snapchat.R

class MainActivity : AppCompatActivity() {

    var emailEditText: EditText? = null
    var passwordEditText: EditText? = null
    val mAuth = FirebaseAuth.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        emailEditText = findViewById(R.id.emailEditText)
        passwordEditText = findViewById(R.id.emailEditText)

        if(mAuth.currentUser != null){
            logIn()
        }
    }

    fun goClicked(view: View){
        // Check if we can log in the user
        mAuth.signInWithEmailAndPassword(emailEditText.text.toString(), passwordEditText.text.toString())
                .addOnCompleteListener(this) { task ->
                    if (task.isSuccessful) {
                        logIn()
                    } else {
                        // Sign up User
                        mAuth.createUserWithEmailAndPassword(emailEditText.text.toString(), passwordEditText.text.toString()).addOnCompleteListener(this){ task ->
                            if(task.isSuccessful){
                                logIn()
                            }
                        }

                    }
                }

        // Sign up the user
    }

    fun logIn(){
        // Move to next Activity
    }
}